crates/astria-sequencer-relayer/src/relayer/write/mod.rs

```diff
@@ -215,6 +236,13 @@ impl BlobSubmitter {
 
                 // add new blocks to the next submission if there is space.
                 Some(block) = self.blocks.recv(), if self.has_capacity() => {
+                    if block.height() <= started_submission.last_submission_sequencer_height() {
+                        info!(
+                            sequencer_height = %block.height(),
+                            "skipping sequencer block as already included in previous submission"
+                        );
+                        continue;
+                    }
```

Comment:
Probably not necessary, but could remove the `continue;` above in favor of merging with the next line, i.e.:
```rust
} else if let Err(error) = self.add_sequencer_block_to_next_submission(block) {`
```

Reply:
Done in [9ade24c](https://github.com/astriaorg/astria/pull/1234/commits/9ade24cf8a1b470ef1ba8524afad2fcbab2dd4ab).